### DISCUSSION TITLE
Why /run/user/0 is owned by the default user?

### DISCUSSION THREAD

**Discussion by davidshen84** [microsoft/WSL#9344] [General] [answered]
Hi,

```
WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.963
```

I have `systemd` enabled on my WSL. My `user@0.service` unit failed because the `/run/user/0` folder is owned by user 1000, not 0.

`/run/user/0` is mounted at runtime, but I could not figure out why it is not set to be owned by root. Should I disable the `user@0.service` unit?

Thanks

**Comment by davidshen84** [comment]
I just learned that in a regular Linux system with `systemd`, the `/run/user/UID` is created **only** if the user is logged in. That means, on my WSL system, the `root` user is logged in, but somehow the `/run/user/0` folder is owned by my default user. I guess it is a special WSL setting?

**Comment by nonylene** [comment]
I found this issue occurs when using Docker Desktop for Windows.

**Comment by nonylene** [comment, marked as answer]
On my environment (Ubuntu 20.04), switching to `docker.io` package on Ubuntu resolves this issue.

**Comment by davidshen84** [comment]
Yes, I use Docker desktop too.

Do you mean you disabled wsl integration and installed docker client in the
wsl manually?


Thanks

**Reply by nonylene** [reply]
I completely uninstalled Docker Desktop from Windows and installed Linux Docker on WSL.

If you need Docker on Windows, you can also disable WSL integration alternatively (⁠you will lose access Docker on Windows from your WSL).

**Reply by davidshen84** [reply]
I don't remember if we could install Docker for Linux in the WSL environment before. Is it supported only recently? But if it is possible, I'd prefer to use the Linux Docker and remove Docker Desktop for Windows.

**Reply by nonylene** [reply]
Sorry I dont know whether Docker supports systemd-enabled WSL2, but on my environment it works well.
